sphinxcontrib/inmanta/api.py

```diff
@@ -435,62 +588,198 @@ def filter_func(name):
                 if filter.match(name):
                     return True
             return False
+
         return filter_func
 
-    def run(self, module_repo: Optional[str], module_name: str, extra_modules: Sequence[str], source_repo: str):
+    def run(
+        self, module_source_dir: str, module_name: str, extra_modules: Sequence[str]
+    ) -> str:
         """
         Run the module doc generation.
 
-        :param module_repo: Absolute path to the directory where all v1 modules are stored. May be None if the main and all
-            extra modules are v2 modules.
+        :param module_source_dir: Absolute path to the directory where the source of the module is.
         :param module_name: The name of the module to generate docs for.
         :param extra_modules: The names of any extra modules.
+
+        :returns: The documentation for this module as a string.
         """
-        mod_data: Optional[Tuple[module.Module, List[str]]] = self._get_modules(module_repo, module_name)
+        mod_data: Optional[Tuple[module.Module, List[str]]] = self._get_modules(
+            module_source_dir, module_name
+        )
         if mod_data is None:
-            raise Exception(f"Could not find module {module_name}.")
+            raise Exception(f"Could not find module {module_name} in {module_source_dir}.")
         mod, submodules = mod_data
 
-        module_filter = self.get_module_filter(None if not module_repo else os.path.join(module_repo, module_name))
+        module_filter = self.get_module_filter(
+            None
+            if not module_source_dir
+            else os.path.join(module_source_dir, module_name)
+        )
 
         for name in extra_modules:
-            extra_mod_data: Optional[Tuple[module.Module, List[str]]] = self._get_modules(module_repo, name)
+            extra_mod_data: Optional[Tuple[module.Module, List[str]]] = (
+                self._get_modules(module_source_dir, name)
+            )
             if extra_mod_data is not None:
                 submodules.extend(extra_mod_data[1])
 
         submodules = sorted([sm for sm in set(submodules) if module_filter(sm)])
         print("Selected sub-modules: " + ", ".join(submodules))
 
-        lines = self.emit_intro(mod, source_repo)
-        lines.extend(self.doc_compile(module_repo, mod.name, submodules))
+        lines = self.emit_intro(mod)
+        lines.extend(self.doc_compile(module_source_dir, mod.name, submodules))
         lines = [line for line in lines if line is not None]
         return "\n".join(lines)
 
-
 @click.command()
 @click.option(
-    "--module_repo",
+    "--module-sources",
     help=(
-        "The repo where all v1 modules are stored (local file). Ignored for v2 modules, which are always fetched from the"
-        " Python environment."
+        "The path to the local directory where all module sources (both v1 and v2) are stored."
```

Comment:
This wording gives the impression that each module has to be present as a v1 and a v2 module.
```suggestion
        "The path to the local directory where all module sources (v1 or v2) are stored."
```